Tophat/Utilities/InstallationTicketMachine.swift

```diff
@@ -96,7 +96,7 @@ struct InstallationTicketMachine {
 
 							let application = try await applicationDownloader.download(from: recipe.source, context: context)
 
-							providedBuildTypes[application.platform, default: []].formUnion(application.targets)
+							await state.track(providedDestinations: application.targets, for: application.platform)
```

Comment:
We used to have a crash here, which made sense.  Multiple tasks were writing to `providedBuildTypes` at the same time.

Swift 6 isn't the easiest to adopt for many existing codebases, but it sure does catch mistakes like this one well!